tsdb/db.go

```diff
@@ -1386,12 +1391,6 @@ func (db *DB) reloadBlocks() (err error) {
 		db.metrics.reloads.Inc()
 	}()
 
-	// Now that we reload TSDB every minute, there is high chance for race condition with a reload
-	// triggered by CleanTombstones(). We need to lock the reload to avoid the situation where
-	// a normal reload and CleanTombstones try to delete the same block.
-	db.mtx.Lock()
-	defer db.mtx.Unlock()
-
 	loadable, corrupted, err := openBlocks(db.logger, db.dir, db.blocks, db.chunkPool)
```

Comment:
> I believe that this reading path requires a lock: https://github.com/prometheus/prometheus/pull/12920/files#diff-495908409affb7da01cb15f2a89b405f105378984919d6149b5cdaaf5e275c7eR1420-R1424

and here.. I believe that certain blocks may temporarily disappear when the lock scope becomes smaller

Reply:
Well spotted, thanks.
Reads on `db.blocks` do require a read lock using `db.mtx`, added.